### DISCUSSION TITLE
TypeError after 3.0 upgrade from 2.11.12

### DISCUSSION THREAD

**Discussion by bp4577** [netbox-community/netbox#7238] [Q&A]
Attempting to perform an upgrade from 2.11.12 to 3.0.1 or 3.0.2 results in the bellow error whenever I attempt to navigate away from the NetBox homepage (/ipam/, /dcim/, ect.). 

<class 'TypeError'>

'<' not supported between instances of 'NoneType' and 'int'

Python version: 3.7.5
NetBox version: 3.0.1

upgrade.sh appears to complete perfectly fine outside of docs build warnings.

**Comment by candlerb** [comment]
It's possible your Netbox is corrupt somehow, as I haven't seen that error reported by anyone else.  Did you install and upgrade using git, or by untarring a release tarball?  Or you may have some garbage in your `configuration.py`

If you can show a full backtrace then that should highlight the problem.  If you set ADMINS and EMAIL in configuration.py then it should be E-mailed to you.  Otherwise, set DEBUG = True and then copy-paste the backtrace from the web interface (please do this as text, not as an image)

One other thing: try

```
python3
>>> import platform
>>> platform.python_version_tuple()
('3', '9', '4')
```

Does this return a sensible value?  I'm wondering whether your problem is triggered by this line of code:

```
netbox/netbox/settings.py:if platform.python_version_tuple() < ('3', '7'):
```

(but then I'd expect the homepage to be broken too)

**Comment by jeremystretch** [comment]
Can you please set `DEBUG = True` in `configuration.py` to enable debugging, restart NetBox, and then provide the complete stack trace?
